src/metabase/query_processor/middleware/resolve_fields.clj

```diff
@@ -14,10 +14,11 @@
 
 (defn resolve-fields
   "Resolve all field referenced in the `query`, and store them in the QP Store."
-  [query]
-  (let [ids (into (set (mbql.u/match (:query query) [:field (id :guard integer?) _] id))
-                  (comp cat (keep :id))
-                  (mbql.u/match (:query query) {:source-metadata source-metadata} source-metadata))]
+  [{:keys [info] :as query}]
+  (let [ids (-> (set (mbql.u/match (:query query) [:field (id :guard integer?) _] id))
+                (into (mbql.u/match (:metadata/dataset-metadata info) [:field (id :guard integer?) _] id))
+                (into (comp cat (keep :id))
+                      (mbql.u/match (:query query) {:source-metadata source-metadata} source-metadata)))]
```

Comment:
Usually instead of threading together `into` forms you can combine them into a single one using transducers and `comp`... since that was how it was in the first place I would suggest keeping it that way and just adding more transducers to the `comp`. I think that would be easier to read than threading together several `into`s